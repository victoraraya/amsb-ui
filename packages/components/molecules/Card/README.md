# `@amsb-ui/card`

> TODO: description

## Usage

```
import Button from '@amsb-ui/card';

// TODO: DEMONSTRATE API
```
