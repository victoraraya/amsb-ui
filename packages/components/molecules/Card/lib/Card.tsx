import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import { CardActionArea, Divider, Link } from '@mui/material';
import { styled } from '@mui/material/styles';
import Typography from '@amsb-ui/typography';
import { AirMilesMediaCardProps } from './types';

const StyledCard = styled(Card)`
    max-width: 28.5rem;
    box-shadow: 0px 1px 8px 0px rgba(16, 24, 40, 0.06);
    outline: 1px solid white;
    :hover {
        outline: 1px solid #61c1ee;
        transition: all ease-in-out 0.3s;
    }
    :focus {
        outline: 1px solid #61c1ee;
        transition: all ease-in-out 0.3s;
    }
`;

const StyledCardActionArea = styled(CardActionArea)`
    .MuiCardActionArea-focusHighlight {
        background: transparent;
    }
`;

const AirMilesMediaCard = ({
    image,
    imageAlt,
    content,
    link,
    linkText,
    target = '_blank',
}: AirMilesMediaCardProps) => (
    <StyledCard>
        <Link href={link} target={target} color="text.secondary" underline="none">
            <StyledCardActionArea>
                <CardMedia
                    component="img"
                    height="68"
                    image={image}
                    alt={imageAlt}
                    sx={{ objectFit: 'none', pt: '1rem', pb: '0.25rem' }}
                />
                <CardContent>
                    <Typography
                        sx={{ mb: '1rem', minHeight: '90px', width: '95%' }}
                        variant="body1"
                    >
                        {content}
                    </Typography>
                    <Divider />
                    <CardActions sx={{ mt: '0.5rem' }}>
                        <Typography variant="subtitle2" color="text.secondary">
                            {linkText}
                        </Typography>
                    </CardActions>
                </CardContent>
            </StyledCardActionArea>
        </Link>
    </StyledCard>
);

export default AirMilesMediaCard;
