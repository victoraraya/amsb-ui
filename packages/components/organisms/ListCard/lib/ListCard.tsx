import Grid from '@amsb-ui/grid';
import Card from '@amsb-ui/card';
import { AirMilesMediaCardProps } from './types';

const ListCard = ({ data }: any) => {
    const sizeValue = data.length <= 2 ? 6 : 4;

    return (
        <Grid container justifyContent="space-between" spacing={4.5}>
            {data.map((card: AirMilesMediaCardProps) => (
                <Grid item xs={12} sm={sizeValue} key={card.id}>
                    <Card
                        image={card.image}
                        imageAlt={card.imageAlt}
                        content={card.content}
                        link={card.link}
                        linkText={card.linkText}
                        id={card.id}
                    />
                </Grid>
            ))}
        </Grid>
    );
};

export default ListCard;
