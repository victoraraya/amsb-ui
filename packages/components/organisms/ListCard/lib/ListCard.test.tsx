import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Card from '@amsb-ui/card';
import ListCard from './ListCard';

const datos = [
    {
        image: 'test',
        imageAlt: 'test',
        content: 'test',
        link: 'test',
        linkText: 'test',
        id: 1,
    },
    {
        image: 'foo',
        imageAlt: 'foo',
        content: 'foo',
        link: 'foo',
        linkText: 'foo',
        id: 2,
    },
];

it('should render successfully', () => {
    const { asFragment } = render(<ListCard data={datos} />);

    const card1 = (
        <Card image="test" imageAlt="test" content="test" link="test" linkText="test" id={1} />
    );
    const card2 = (
        <Card image="foo" imageAlt="foo" content="foo" link="foo" linkText="foo" id={2} />
    );

    expect(card1).toBeDefined();
    expect(card2).toBeDefined();

    expect(asFragment()).toMatchSnapshot();
});
