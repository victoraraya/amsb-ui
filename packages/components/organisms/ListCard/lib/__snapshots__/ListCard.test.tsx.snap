// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render successfully 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -36px;
  width: calc(100% + 36px);
  margin-left: -36px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-0>.MuiGrid-item {
  padding-top: 36px;
}

.emotion-0>.MuiGrid-item {
  padding-left: 36px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
}

@media (min-width:600px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:900px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:1536px) {
  .emotion-1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

.emotion-2 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  max-width: 28.5rem;
  box-shadow: 0px 1px 8px 0px rgba(16, 24, 40, 0.06);
  outline: 1px solid white;
}

.emotion-2:hover {
  outline: 1px solid #61c1ee;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
}

.emotion-2:focus {
  outline: 1px solid #61c1ee;
  -webkit-transition: all ease-in-out 0.3s;
  transition: all ease-in-out 0.3s;
}

.emotion-3 {
  margin: 0;
  color: rgba(0, 0, 0, 0.6);
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: block;
  text-align: inherit;
  width: 100%;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4:hover .MuiCardActionArea-focusHighlight {
  opacity: 0.04;
}

@media (hover: none) {
  .emotion-4:hover .MuiCardActionArea-focusHighlight {
    opacity: 0;
  }
}

.emotion-4.Mui-focusVisible .MuiCardActionArea-focusHighlight {
  opacity: 0.12;
}

.emotion-4 .MuiCardActionArea-focusHighlight {
  background: transparent;
}

.emotion-5 {
  display: block;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  -webkit-background-position: center;
  background-position: center;
  width: 100%;
  object-fit: cover;
  object-fit: none;
  padding-top: 1rem;
  padding-bottom: 0.25rem;
}

.emotion-6 {
  padding: 16px;
}

.emotion-6:last-child {
  padding-bottom: 24px;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  margin-bottom: 1rem;
  min-height: 90px;
  width: 95%;
}

.emotion-8 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  margin-top: 0.5rem;
}

.emotion-9>:not(:first-of-type) {
  margin-left: 8px;
}

.emotion-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.57;
  letter-spacing: 0.00714em;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-11 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
  opacity: 0;
  background-color: currentcolor;
  -webkit-transition: opacity 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-12 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4.5 emotion-0"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 emotion-1"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-2"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone emotion-3"
          href="test"
          target="_blank"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root emotion-4"
            tabindex="0"
            type="button"
          >
            <img
              alt="test"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img emotion-5"
              height="68"
              src="test"
            />
            <div
              class="MuiCardContent-root emotion-6"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 emotion-7"
              >
                test
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth emotion-8"
              />
              <div
                class="MuiCardActions-root MuiCardActions-spacing emotion-9"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 emotion-10"
                >
                  test
                </h6>
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight emotion-11"
            />
            <span
              class="MuiTouchRipple-root emotion-12"
            />
          </button>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 emotion-1"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-2"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone emotion-3"
          href="foo"
          target="_blank"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root emotion-4"
            tabindex="0"
            type="button"
          >
            <img
              alt="foo"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img emotion-5"
              height="68"
              src="foo"
            />
            <div
              class="MuiCardContent-root emotion-6"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 emotion-7"
              >
                foo
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth emotion-8"
              />
              <div
                class="MuiCardActions-root MuiCardActions-spacing emotion-9"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 emotion-10"
                >
                  foo
                </h6>
              </div>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight emotion-11"
            />
            <span
              class="MuiTouchRipple-root emotion-12"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
